From 458b6001fda7662713ff0275319e6ab458588819 Mon Sep 17 00:00:00 2001
From: Dongwon Kim <dongwon.kim@intel.com>
Date: Tue, 28 Feb 2023 20:47:04 -0800
Subject: [PATCH 43/44] ui/gtk: additional unblocking condition in window
 iconized event

Current routine does not unblock the pipeline if the GTK window is minimized
before actual rendering is finished. Fence_id is >= 0 in this case so this
check should be added as one of unblocking conditions.

Signed-off-by: Dongwon Kim <dongwon.kim@intel.com>
---
 ui/gtk.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/ui/gtk.c b/ui/gtk.c
index 6adf1c9097..f0f5115550 100644
--- a/ui/gtk.c
+++ b/ui/gtk.c
@@ -1491,7 +1491,8 @@ static gboolean gd_window_state_event(GtkWidget *widget, GdkEvent *event,
         vc->gfx.visible = false;
         gd_set_ui_size(vc, 0, 0);
         if (vc->gfx.guest_fb.dmabuf &&
-            vc->gfx.guest_fb.dmabuf->draw_submitted) {
+            (vc->gfx.guest_fb.dmabuf->fence_fd >= 0 ||
+            vc->gfx.guest_fb.dmabuf->draw_submitted)) {
             vc->gfx.guest_fb.dmabuf->draw_submitted = false;
             graphic_hw_gl_block(vc->gfx.dcl.con, false);
         }
-- 
2.20.1

